cpp/src/Ice/ServantManager.cpp

```diff
@@ -65,18 +65,18 @@ IceInternal::ServantManager::addDefaultServant(const shared_ptr<Object>& object,
         throw AlreadyRegisteredException(__FILE__, __LINE__, "default servant", category);
     }
 
-    _defaultServantMap.insert(pair<const string, shared_ptr<Object>>(category, object));
+    _defaultServantMap.insert(pair<const string, ObjectPtr>(category, object));
 }
 
-shared_ptr<Object>
+ObjectPtr
 IceInternal::ServantManager::removeServant(const Identity& ident, const string& facet)
 {
     //
     // We return the removed servant to avoid releasing the last reference count
     // with *this locked. We don't want to run user code, such as the servant
     // destructor, with an internal Ice mutex locked.
     //
-    shared_ptr<Object> servant = 0;
+    ObjectPtr servant = 0;
```

Comment:
```suggestion
    ObjectPtr servant = nullptr;
```

Reply:
The correct fix is to not copy-initialize it at all.

Reply:
Fixed. Didn't need to construct it there at all.